tests/orion/api/test_deployments.py

```diff
@@ -269,6 +269,45 @@ async def test_upserting_deployment_with_new_schedule_deletes_existing_auto_sche
         result = await session.execute(query)
         assert result.scalar() < pendulum.now().add(seconds=100)
 
+    async def test_create_deployment_throws_useful_error_on_missing_blocks(
+        self,
+        client,
+        flow,
+        infrastructure_document_id,
+        storage_document_id,
+    ):
+        data = DeploymentCreate(
+            name="My Deployment",
+            manifest_path="file.json",
+            flow_id=flow.id,
+            tags=["foo"],
+            parameters={"foo": "bar"},
+            infrastructure_document_id=uuid4(),
+            storage_document_id=storage_document_id,
+        ).dict(json_compatible=True)
+        response = await client.post("/deployments/", json=data)
+        assert response.status_code == status.HTTP_409_CONFLICT
+        assert (
+            "Error creating deployment. Could not find infrastructure block with id"
+            in response.json()["detail"]
+        ), "Error message identifies infrastructure block could not be found"
+
+        data = DeploymentCreate(
+            name="My Deployment",
+            manifest_path="file.json",
+            flow_id=flow.id,
+            tags=["foo"],
+            parameters={"foo": "bar"},
+            infrastructure_document_id=infrastructure_document_id,
+            storage_document_id=uuid4(),
+        ).dict(json_compatible=True)
+        response = await client.post("/deployments/", json=data)
+        assert response.status_code == status.HTTP_409_CONFLICT
+        assert (
+            "Error creating deployment. Could not find storage block with id"
+            in response.json()["detail"]
+        ), "Error message identifies storage block could not be found."
```

Comment:
this is just a note -- the message here will be raised if the `assert` statement fails, so this could be a confusing message to see when the error message _doesn't_ identify the storage block